test_fms/sat_vapor_pres/test_sat_vapor_pres.F90

```diff
@@ -0,0 +1,1009 @@
+!*
+!* FMS is distributed in the hope that it will be useful, but WITHOUT
+!* ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+!* FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License
+!* for more details.
+!*
+!* You should have received a copy of the GNU Lesser General Public
+!* License along with FMS.  If not, see <http://www.gnu.org/licenses/>.
+!***********************************************************************
+!> @file
+!! @brief unit test for the mpp_root_pe() function
+!! @author MiKyung Lee
+!! @email gfdl.climate.model.info@noaa.gov
+!! @description This program tests mainly the lookup* procedures in sat_vapor_pres_mod.
+!!              The compute_tables, compute_es_k, and compute_es_liq_k subroutines found in
+!!              this file are copied from sat_vapor_pres_k.F90 in order to generate answers.
+!! TODO:  A more comprehensive testing suite for the subroutine compute_qd
+!! TODO:  A more comprehensive testing suite for the compute_mrs
+!! TODO:  A test to check computation of TABLE and DTABLE when do_simple=.true.
+!!        (see subroutine sat_vapor_press_k_init)
+!! TODO:  Testing suite to test computations involving D2TABLE, D2TABLE2, D2TABLE3
+!!        Current tests for the lookup* subroutines only checks esat and desat for temperatures
+!!        = TCMIN and TCMAX.  The D2* tables are not involved in the computation for these two cases.
+!!        Thus, testing suite that involves these D2* table values should be incorporated here.
+!! TODO:  Test the computation of nbads and test to to see if the expected error occurs if the temperature
+!!        is less than TCMIN or higher than TCMAX
+
+program test_sat_vap_pressure
+
+use            fms_mod, only: fms_init, fms_end
+use            mpp_mod, only: mpp_error, FATAL
+use       platform_mod, only: r4_kind, r8_kind
+use      constants_mod, only: RDGAS, RVGAS, TFREEZE
+use sat_vapor_pres_mod, only: TCMIN, TCMAX, sat_vapor_pres_init, &
+                              compute_qs, compute_mrs,           &
+                              lookup_es,   lookup_des,  lookup_es_des,   &
+                              lookup_es2,  lookup_des2, lookup_es2_des2, &
+                              lookup_es3,  lookup_des3, lookup_es3_des3
+
+implicit none
+
+integer, parameter :: ESRES=10  !> taken from sat_vapor_pres_mod
+real(r8_kind), dimension(:), allocatable :: TABLE, DTABLE, TABLE2, DTABLE2, TABLE3, DTABLE3
+integer :: io, N
+
+character(100) :: nml_file
+logical :: test1, test2, test3, test4, test5
+NAMELIST / test_sat_vapor_pres_nml/ test1, test2, test3, test4, test5
+
+N=(TCMAX-TCMIN)*ESRES+1
+allocate( TABLE(N),DTABLE(N),TABLE2(N),DTABLE2(N),TABLE3(N),DTABLE3(N) )
+
+call fms_init()
+call sat_vapor_pres_init()  !> compute tables to be used for testing
+call compute_tables()       !> compute tables to generate answers/reference values
+
+nml_file='test_sat_vapor_pres.nml'
+open(unit=100, file=trim(nml_file), action='read')
```

Comment:
I'd suggest `newunit=nml_unit_var` instead of `unit=constant`, since variable names are more descriptive and easier to remember than numbers.